packages/squiggle-lang/src/ast/peggyParser.peggy

```diff
@@ -397,36 +397,36 @@ dictConstructor 'dict'
 
 // Separators
 
+statementSeparator ';'
+  = (_ ';' / _ newLineWithPossibleComment)+ _nl
+
+commaSeparator ',' = _ ',' _nl
+
+// optional whitespace without line breaks (line breaks are still allowed inside block comments)
 _ 'whitespace'
   = whiteSpaceCharactersOrComment*
 
+// optional whitespace with possible line breaks and comments
 _nl 'whitespace'
-  = (whiteSpaceCharactersOrComment / commentOrNewLine)*
-
-__ 'whitespace'
-  = whiteSpaceCharactersOrComment+
+  = (whiteSpaceCharactersOrComment / newLineWithPossibleComment)*
 
+// required whitespace with possible line breaks and comments
 __nl 'whitespace'
-  = (whiteSpaceCharactersOrComment / commentOrNewLine)+
-
-statementSeparator ';'
-  = _ (';' / commentOrNewLine)+ _nl
-
-commaSeparator ',' = _ ',' _nl
+  = (whiteSpaceCharactersOrComment / newLineWithPossibleComment)+
 
-commentOrNewLine = finalComment? newLine
+newLineWithPossibleComment = lineComment? newLine
   newLine "newline" = [\n\r]
 
-finalComment "line comment"
-  = _ ('//') comment:($([^\r\n]*))
-    { options.comments.push(h.lineComment(comment, location())); }
+lineComment "line comment"
+  = '//' comment:($([^\r\n]*))
```

Comment:
The changes to `statementSeparator` above are important because I had to remove the first `_` from this rule. Otherwise, comments were duplicated even with deduplication code (once it was "  // comment" and once "// comment", different locations -> no deduplication).